tests/jballerina-unit-test/src/test/resources/test-src/klass/class_def_test.bal

```diff
@@ -31,6 +31,22 @@ function testFooClass() {
     assertEquality(f.s, COMMA);
 }
 
+int a = 3;
+
+public class Bar {
+    int a = 4;
+    int b = a;
+
+    function init() {
+        assertEquality(self.b, 3);
+    }
+}
+
+function testGlobalVariablesAssignmentToFields() {
+    Bar b = new();
+    assertEquality(b.b, 3);
```

Comment:
Shall we also update the module-level variable and test?

```ballerina
    a = 6;
    assertEquality(b.b, 3);
    Bar b2 = new;
    assertEquality(b2.b, 6);
```

Reply:
Added